ci/test/06_script_b.sh

```diff
@@ -36,7 +36,7 @@ fi
 
 if [ "${RUN_TIDY}" = "true" ]; then
   export P_CI_DIR="${BASE_BUILD_DIR}/bitcoin-$HOST/src/"
-  CI_EXEC run-clang-tidy "${MAKEJOBS}"
+  ( CI_EXEC run-clang-tidy -quiet "${MAKEJOBS}" ) | grep -C5 "error"
```

Comment:
It would be good to test this to see at least one output example and check that the `|` doesn't eat the exit code

Reply:
Pushed up a commit that should error

Reply:
[Error output is showing](https://cirrus-ci.com/task/5035600825810944?logs=ci#L777):
```bash
clang-tidy-14 --use-color -p=/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu -quiet /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/bitcoin-cli.cpp
/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/bitcoin-cli.cpp:70:17: error: redundant string initialization [readability-redundant-string-init,-warnings-as-errors]
    std::string blah = "";
                ^~~~~~~~~
                blah
```

Reply:
Yeah, but the exit code is 0 (green), when it used to be 1 (red)

Reply:
Perhaps we should use `set -eo pipefail` for these scripts?

Reply:
> Perhaps we should use set -eo pipefail for these scripts?

Looks like that will acheive what we want